packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -798,22 +796,42 @@ function FirstTermDiscountCallout( { product }: { product: ResponseCartProduct }
 		// Do not display discount reason if a coupon is applied.
 		isCouponApplied( product )
 	) {
-		return null;
+		return <></>;
 	}
-
+	let oneTimeExplanation: ReactChild = '';
 	if ( isMonthlyProduct( product ) ) {
-		return <DiscountCallout>{ translate( 'Discount for first month' ) }</DiscountCallout>;
+		oneTimeExplanation = translate( 'Discount: %(discount)s Applied in first month only', {
+			args: {
+				discount: formatCurrency( discount, product.currency, {
+					isSmallestUnit: true,
+					stripZeros: true,
+				} ),
+			},
+		} );
 	}
 
 	if ( isYearly( product ) ) {
-		return <DiscountCallout>{ translate( 'Discount for first year' ) }</DiscountCallout>;
+		oneTimeExplanation = translate( 'Discount: %(discount)s Applied in first year only', {
```

Comment:
I'm not certain, but I think that "applied" should not be capitalized since it's part of an ongoing sentence.